roles/download/tasks/download_file.yml

```diff
@@ -47,11 +47,40 @@
     - download_force_cache
     - not download_localhost
 
+  # We check a number of mirrors that may hold the file and pick a working one at random
+  # This task will avoid logging it's parameters to not leak environment passwords in the log
```

Comment:
Is it possible to see which URL doesn't work to know the URL needs to be updated?

Reply:
Not with the current logic since we put the `no_log` flag on the download task, the error is obscured, including any parameters like the URL. I could add a second task to log the results though that also has some potential to leak credentials if they are embedded in the url.